dcos_installer/upgrades.py

```diff
@@ -0,0 +1,112 @@
+"""Generates a bash script for upgrading by hand """
+
+import gen.installer.util as util
+import gen.template
+from dcos_installer.config import Config
+from dcos_installer.constants import CONFIG_PATH
+from pkgpanda.util import (if_exists, load_json, write_string)
+
+
+node_upgrade_template = """
+#!/bin/bash
+#
+# BASH script to upgrade DC/OS on a node
+#
+# Metadata:
+#   dcos image commit: {{ dcos_image_commit }}
+#   generation date: {{ generation_date }}
+# Copyright 2016 Mesosphere, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit -o nounset -o pipefail
+
+# Check if this is a terminal, and if colors are supported, set some basic
+# colors for outputs
+if [ -t 1 ]; then
+    colors_supported=$(tput colors)
+    if [[ $colors_supported -ge 8 ]]; then
+    RED='\e[1;31m'
+    BOLD='\e[1m'
+    NORMAL='\e[0m'
+    fi
+fi
+
+if [[ $EUID -ne 0 ]]; then
+    echo "This script must be run as root" 1>&2
+    exit 1
+fi
+
+function check_command_exists() {
+    COMMAND=$1
+    DISPLAY_NAME=${2:-$COMMAND}
+
+    echo -e -n "Checking if $DISPLAY_NAME is installed and in PATH: "
+    $( command -v $COMMAND >/dev/null 2>&1 || exit 1 )
+    RC=$?
+    print_status $RC
+    (( OVERALL_RC += $RC ))
+    return $RC
+}
+
+function main() {
+    # source the environment required to run the pkgpanda commands
+    source /opt/mesosphere/environment.export
+
+    # check we get pkgpanda once we have the environment setup
+    check_command_exists pkgpanda
+
+    {{ pkgpanda_fetch }}
+
+    {{ pkgpanda_activate }}
+}
```

Comment:
We'll at least be moving the ZooKeeper data dir by moving and/or symlinking directories and making requests against the Exhibitor UI.